packages/cli/src/util/build/write-build-result.ts

```diff
@@ -371,6 +371,13 @@ async function writeLambda(
   }
 
   const memory = functionConfiguration?.memory ?? lambda.memory;
+
+  if (functionConfiguration?.maxDuration && lambda.maxDuration) {
+    throw new Error(
+      `Cannot specify both \`maxDuration\` in \`vercel.json\` and \`${path}\``
```

Comment:
Defining in both the `vercel.json` and config seems to not throw an error but silently have the value in `vercel.json` take precedent. Should we retain this behavior or diff between those settings and throw an error?

Reply:
I don't think we should have this throw. It's a breaking change. We should document that the `vercel.json` overrides the function's config.

I advise against logging a warning in the event both are defined as it could make the output noisy. Documentation should clearly describe the behavior.

If we were to introduce any breaking change, I would have the function's config override the `vercel.json` and document this change.

Reply:
I think we should swap the order actually. The `lambda` value would take precedence. Reason being that the `lambda` value presumably comes from the `export const config` which is more specific/explicit (individual file level) than the `vercel.json` which is a glob pattern. Plus we are trying to encourage config-in-code more anyways.

This would technically be a breaking change, but since there is currently no way to define those configs on `lambda` anyways aside from `runtime`, it would be net-new behavior and thus non-breaking.

Reply:
Agree with @TooTallNate

Reply:
Retaining the existing behavior in this PR, let's ship another PR with that breaking change of having the lambda config take priority?